R/merge.R

```diff
@@ -50,11 +50,30 @@ merge.data.table = function(x, y, by = NULL, by.x = NULL, by.y = NULL, all = FAL
       by = intersect(nm_x, nm_y)
     if (length(by) == 0L || !is.character(by))
       stopf("A non-empty vector of column names for `by` is required.")
-    if (!all(by %chin% intersect(nm_x, nm_y)))
-      stopf("Elements listed in `by` must be valid column names in x and y")
-    by = unname(by)
-    by.x = by.y = by
+    # UPDATED PART STARTS HERE
```

Comment:
This comment will stop being helpful after the suggested change is accepted, and we have more reliable ways of finding out which parts of the code are updated.